flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/source/TestIcebergSourceWithWatermarkExtractor.java

```diff
@@ -383,8 +381,10 @@ protected IcebergSource<RowData> source() {
         .project(TestFixtures.TS_SCHEMA)
         .splitSize(100L)
         .streaming(true)
-        .monitorInterval(Duration.ofMillis(2))
+        .monitorInterval(Duration.ofMillis(100))
```

Comment:
Added this, because the 2ms will causing too much load and ended up causing flakiness:
```
[collect-sink-operator-coordinator-executor-thread-pool-thread-1] WARN org.apache.flink.streaming.api.operators.collect.CollectSinkOperatorCoordinator - Collect sink coordinator encounters a CompletionException: java.net.SocketTimeoutException: Read timed out
```

Reply:
I run the tests during the whole day yesterday, and found that the previous fix removed most of the flakiness, but we still have some issues which fails around 1000-2000 runs.

In failure cases the records are emitted by the emitter, but none of the are arriving to the Sink. I see the above error in the logs in this case: `SocketTimeout